lib/rubocop/cop/style/redundant_regexp_argument.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Identifies places where argument can be replaced from
+      # a deterministic regexp to a string.
+      #
+      # @example
+      #   # bad
+      #   'foo'.byteindex(/f/)
+      #   'foo'.byterindex(/f/)
+      #   'foo'.gsub(/f/, 'x')
+      #   'foo'.gsub!(/f/, 'x')
+      #   'foo'.index(/f/)
+      #   'foo'.partition(/f/)
+      #   'foo'.rindex(/f/)
+      #   'foo'.rpartition(/f/)
+      #   'foo'.scan(/f/)
+      #   'foo'.split(/f/)
+      #   'foo'.sub(/f/, 'x')
+      #   'foo'.sub!(/f/, 'x')
+      #
+      #   # good
+      #   'foo'.byteindex('f')
+      #   'foo'.byterindex('f')
+      #   'foo'.gsub('f', 'x')
+      #   'foo'.gsub!('f', 'x')
+      #   'foo'.index('f')
+      #   'foo'.partition('f')
+      #   'foo'.rindex('f')
+      #   'foo'.rpartition('f')
+      #   'foo'.scan('f')
+      #   'foo'.split('f')
+      #   'foo'.sub('f', 'x')
+      #   'foo'.sub!('f', 'x')
+      class RedundantRegexpArgument < Base
+        extend AutoCorrector
+
+        MSG = 'Use string `%<prefer>s` as argument instead of regexp `%<current>s`.'
+        RESTRICT_ON_SEND = %i[
+          byteindex byterindex gsub gsub! index partition rindex rpartition scan split sub sub!
+        ].freeze
+        DETERMINISTIC_REGEX = /\A(?:#{LITERAL_REGEX})+\Z/.freeze
+        STR_SPECIAL_CHARS = %w[\n \" \' \\\\ \t \b \f \r].freeze
+
+        def on_send(node)
+          return unless (regexp_node = node.first_argument)
+          return unless regexp_node.regexp_type?
+          return if regexp_node.ignore_case? || regexp_node.content == ' '
```

Comment:
i'm not sure what `regexp_node.content == ' '` implies, but maybe `''` was intended instead of `' '`, to allow something like `split(//)`?

Reply:
It prevents false positives in regex matches for spaces.

```ruby
'foo         bar'.split(/ /) # => ["foo", "", "", "", "", "", "", "", "", "bar"]
'foo         bar'.split(' ') # => ["foo", "bar"]
```

The test is written below for that.
https://github.com/rubocop/rubocop/pull/11595/files#diff-86f0d8dfae0d2d9eb2236fbfd1ae5c61165805156ed3c04b0847e6f82823252eR121-R125

Reply:
Ah, i had no idea that split had a custom "awk-like" behavior for single spaces!